zebrad/tests/acceptance.rs

```diff
@@ -4,195 +4,292 @@
+    let mut child_unwrapped = child.unwrap();
+    child_unwrapped.kill()?;
+
+    let output = child_unwrapped.wait_with_output()?;
+    let output = output.assert_failure()?;
 
     // Todo: maybe add special info!() to seed command
     // Todo: improve the regex
-    cmd.stdout()
-        .expect_regex(r"^(.*?)Initializing tracing endpoint");
+    output.stdout_contains(r"^(.*?)Initializing tracing endpoint")?;
 
-    cmd.wait().unwrap_err();
-    // Problem: Zombie child process is not killed
-    //cmd.kill();
+    Ok(())
 }
 
 #[test]
-fn seed_args() {
-    let mut runner = RUNNER.clone();
-    let cmd = runner.args(&["seed", "argument"]).capture_stdout().run();
-    cmd.wait().unwrap().expect_code(1);
+fn seed_args() -> Result<()> {
+    zebra_test::init();
+
+    let child = get_child_multi_args(&["seed", "argument"]);
+    let output = child.unwrap().wait_with_output()?;
+    let output = output.assert_failure()?;
+
+    output.stdout_contains(r"unexpected free argument `argument`")?;
+
+    let child = get_child_multi_args(&["seed", "-f"]);
+    let output = child.unwrap().wait_with_output()?;
+    let output = output.assert_failure()?;
+
+    output.stdout_contains(r"unrecognized option `-f`")?;
+
+    let child = get_child_multi_args(&["seed", "start"]);
+    let output = child.unwrap().wait_with_output()?;
+    let output = output.assert_failure()?;
 
-    let mut runner = RUNNER.clone();
-    let cmd = runner.args(&["seed", "-f"]).capture_stdout().run();
-    cmd.wait().unwrap().expect_code(1);
+    output.stdout_contains(r"unexpected free argument `start`")?;
 
-    let mut runner = RUNNER.clone();
-    let cmd = runner.args(&["seed", "start"]).capture_stdout().run();
-    cmd.wait().unwrap().expect_code(1);
+    Ok(())
 }
 
 #[test]
 fn start_no_args() -> Result<()> {
     zebra_test::init();
 
-    let (mut cmd, _guard) = test_cmd(env!("CARGO_BIN_EXE_zebrad"))?;
-
-    let mut child = cmd
-        .arg("start")
-        .stdout(Stdio::piped())
-        .stderr(Stdio::piped())
-        .spawn2()?;
+    let child = get_child_single_arg("start");
 
+    // Run the program and kill it at 1 second
     std::thread::sleep(Duration::from_secs(1));
+    let mut child_unwrapped = child.unwrap();
+    child_unwrapped.kill()?;
 
-    child.kill()?;
-
-    let output = child.wait_with_output()?;
+    let output = child_unwrapped.wait_with_output()?;
     let output = output.assert_failure()?;
 
+    // Todo: maybe add special info!() to seed command
+    // Todo: improve the regex
     output.stdout_contains(r"^(.*?)Initializing tracing endpoint")?;
 
     Ok(())
 }
 
 #[test]
-fn start_args() {
+fn start_args() -> Result<()> {
+    zebra_test::init();
+
     // Bug? This should fail but not happening
-    //let mut runner = RUNNER.clone();
-    //let cmd = runner.args(&["start", "argument"]).capture_stdout().run();
-    //cmd.wait().unwrap().expect_code(1);
+    //let child = get_child_multi_args(&["start", "argument"]);
```

Comment:
Whats wrong with this test case rn?

Reply:
I think it should be an error however zebra is starting.

Reply:
More at https://github.com/ZcashFoundation/zebra/pull/690#discussion_r458405609

Reply:
https://github.com/ZcashFoundation/zebra/pull/690#discussion_r458981570